R/ops.R

```diff
@@ -123,10 +123,19 @@ st_apply.stars = function(X, MARGIN, FUN, ..., CLUSTER = NULL, PROGRESS = FALSE,
 				} else
 					parallel::parApply(CLUSTER, X = y, MARGIN = MARGIN, FUN = FUN, ...)
 			}
-		if (is.array(ret))
-			ret
-		else
-			array(ret, dX)
+		if (!is.array(ret))
+		{
+			ret <- array(ret,dX)
+		}
+		if(inherits(y,'units')){
+			# calculate function directly on one subset to see what the output units are
+			funUnits<-FUN(do.call('[',c( list(y),ifelse(1:length(dim(y)) %in% MARGIN,1,lapply(dim(y), seq)))))
```

Comment:
Hard to read code! But if I understand it correctly, in the worst case you're redoing an expensive computation just to find out the units of the result.

Reply:
Also, I'd wrap it in a `try()` so it won't break on errors.

Reply:
The idea is to have just run `FUN` for the first combination of margins, this might or might not be expensive depending on the exact dimensions of the data. The `do.call` statement does subset `y` to a similar subset as is done by `apply`. I did not directly find a better way to do this subsetting since the number of dimensions and the margins used are both variable. If there is a better way to subset I'm open to suggestions.

My guess was that this would generally be ok in computational effort.